### PR TITLE
Upgrade to 0.28 - Initial adaptation

### PR DESCRIPTION
**Pull request by entantoencuanto**
Related to #251

This PR is a step in the adaptation of DecidimAwesome to Decidim 0.28 version, which has important changes on the front. At the moment this PR focuses on the parts of the gem that require less of the front-end changes introduced by the module, such as the Fullscreen Iframe and Awesome map components or the menu tweaks or admin assignations to specific scopes

This PR:

* Makes the essential changes in the gem to make it work in the development environment with a version of Decidim that uses the develop branch (preparing 0.28 release):
  * Changing dependencies in the Gemfile
  * Moving the changes from ProposalMCellOverride (m-cells are deprecated in 0.28) to proposals l-cells. These modifications are pending redesign.
  * Temporarily disables the module's custom js and css
* Adapts blank_component controller to work with 0.28 release adding a method for breadcrumbs
* Adapts Fullscreen Iframe component to 0.28
Screenshot from 2023-11-20 11-18-50

* Adapts Awesome Maps component to 0.28

Screenshot from 2023-11-20 16-02-11

* Adapts the Menu Hacks feature to 0.28 and overrides breadcrumb menu item presenter to allow the use of links opened in a new window. It also allows to edit the content block main menu which may appear in the homepage

Screenshot from 2023-12-05 18-00-07
Screenshot from 2023-12-05 18-03-08

* Adapts the modals used to create and edit constraints to decidim_modals introduced in 0.28
* Recovers the custom css code applied by scopes defined by admins in awesome menu
* Adapts the scopes custom css admin management

image
 
* Adapts the scopes admins tweaks admin management

image (1)

* Updates the icons used by the module to use existing declared ones or register new icons specific of the module
* Refactors the secondary nav menu of the module to use the same conventions of Decidim 0.28
* Changes the tests of the adapted features both in admin and front
* Adds skips to a lot of tests related with functionalities not adapted yet to 0.28
* Copies current configuration of rubocop in Decidim 0.28 and fixes the existing offenses
* Renames the 0.27 workflows to 0.28 and removes the 0.26 workflow

Pending:
* [ ] See #257
* [ ] Linters: There is currently a problem trying to get js and styles linters configuration because the release 0.28 does not exist and using [references to 0.27](https://github.com/decidim-ice/decidim-module-decidim_awesome/pull/253/commits/c87b2fca21b2794f5d36e85ff2437e68f02c02c4) seems to break the compilation of the application in the CI engine
* [x] Tests: Remove references in tests to deprecated legacy versions

image (4)

NOTE - This is a currently work in progress, take in account that this module adapts some parts to 0.28 and disables some other so currently there will be a lot of broken features

### CODE REVIEW

**Comment by entantoencuanto** [comment]
Hi, @microstudi, I've added the commented updates to Menu Hacks feature and now admins are able to edit both the main menu displayed in the breadcrumb and footer and the homepage content block menu.

Please, take a look to the PR, it's ready to review the changes covered by it. There is still the linters issue that will not be resolved until version 0.28 is released, because there must be a reference to a version of @decidim/eslint-config and @decidim/stylelint-config compatible with the other dependencies and the last existing version (0.27.4) is not.

**Comment by furilo** [comment]
@microstudi any feedback? ;)

**Comment by paarals** [comment]
> @microstudi any feedback? ;)

morning @furilo, @antopalidi is going to do a first look and then @microstudi during the week. we have some projects to close, so we will try our best. thanks for the work!

File: app/controllers/decidim/decidim_awesome/iframe_component/iframe_controller.rb [hunk 1]
```diff
@@ -20,13 +21,18 @@ def sanitize(html)
           sanitizer.sanitize(html, tags: %w(iframe), attributes: ALLOWED_ATTRIBUTES)
         end
 
-        def remove_margins?
```
Comment (microstudi) [inline app/controllers/decidim/decidim_awesome/iframe_component/iframe_controller.rb:23]: Jus a question here, there are no margins in the new design?

**Review by microstudi — COMMENTED** [review]
Hi @entantoencuanto 
this is an amazing job.
I'm leaving some comments so you can work on them while I look the rest. As v0.28 is out, maybe we can fix the linters now.
Thanks

File: app/controllers/decidim/decidim_awesome/iframe_component/iframe_controller.rb (hunk 1 above)
Comment (entantoencuanto) [reply app/controllers/decidim/decidim_awesome/iframe_component/iframe_controller.rb:23]: No, the section containing the iframe uses the special `html-block-html` id which uses the full width with no margins

image

**Comment by entantoencuanto** [comment]
Humm, it seems that the rubocop configuration is not the same as the used in Decidim, for example the Capybara/NegationMatcher criteria is the opposite, in Decidim is expected to use `expect(...).not_to have_content` and there is expected to use `expect(...).to have_no_content`. I've missed something

**Comment by takahashim** [comment]
@entantoencuanto Hi, `rubocop-capybara` gem's version in Gemfile.lock of this repo is different from one in Decidim.

https://github.com/decidim/decidim/blob/39541c06c0dea58bd2ff16b8f0c55fffeb0520e2/Gemfile.lock#L701

It seems that most of the offenses will disappear if you modify the Gemfile.lock to use `rubocop-capybara` 2.18.0 instead of 2.20.0.

```diff
diff --git a/Gemfile.lock b/Gemfile.lock
index 7c75ea8..cd372e7 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -654,7 +656,7 @@ GEM
       unicode-display_width (>= 2.4.0, < 3.0)
     rubocop-ast (1.30.0)
       parser (>= 3.2.1.0)
-    rubocop-capybara (2.20.0)
+    rubocop-capybara (2.18.0)
       rubocop (~> 1.41)
     rubocop-factory_bot (2.25.1)
       rubocop (~> 1.41)
```

**Comment by entantoencuanto** [comment]
Thanks a lot, @takahashim, you are right!

**Pull request merged**